### ISSUE TITLE
Self Hosted / Docker `supabase/postgres:15.1.0.109_arm64` : Worker does not execute requests

### ISSUE THREAD

**Issue by bentron2000** [supabase/pg_net#98]
- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug
Docker install of `supabase/postgres:15.1.0.109_arm64` 
`pg_net` installs successfully, creates `net` schema etc, and functions seemingly operate as expected, however requests do not execute. They do not time out either.

## To Reproduce

1. install postgres docker image `supabase/postgres:15.1.0.109_arm64` 
2. create a database
3. execute `create extension pg_net;`
4. Send an example query as per readme:
```
SELECT net.http_get (
    'https://postman-echo.com/get?foo1=bar1&foo2=bar2'
) AS request_id;
```
5. query: `SELECT * FROM net.http_request_queue` => responds with one row corresponding to the request
6. query: `SELECT * FROM net._http_response` => responds with no rows
7. `SELECT net.check_worker_is_up()` reports: `ERROR:  the pg_net background worker is not up`
8. manually querying `SELECT * from pg_stat_activity` shows a worker is running - but the `backend_type` from this query reports `"pg_net 0.7.1 worker"` - where the query in the `check_worker_is_up()` function expects `"pg_net worker"`

At this point the request stays in `net.http_request_queue` and never times out etc...

NB. A test hosted on supabase sends requests successfully and populates `net._http_response` with a response, but also shows `ERROR:  the pg_net background worker is not up` when `check_worker_is_up()` is executed.

## Expected behavior

The request should execute, the row in the request table should no longer be present and a response should be in the response table.

## Screenshots

<img width="687" alt="Screenshot 2023-08-07 at 12 41 09 pm" src="https://github.com/supabase/pg_net/assets/5344375/645c46ee-b17f-4f9e-a10d-7cf1fac5c52f">
<img width="753" alt="Screenshot 2023-08-07 at 12 40 40 pm" src="https://github.com/supabase/pg_net/assets/5344375/65598de2-8703-4da3-835b-32a9591b8fcb">
<img width="771" alt="Screenshot 2023-08-07 at 12 38 31 pm" src="https://github.com/supabase/pg_net/assets/5344375/b4dcfa1b-519d-42a9-b6a5-c3c92dbf4698">
<img width="764" alt="Screenshot 2023-08-07 at 12 38 00 pm" src="https://github.com/supabase/pg_net/assets/5344375/e870fde5-e66a-47b1-9bc7-ff3dbbe43826">

## System information

- OS: macOS - running docker instance `supabase/postgres:15.1.0.109_arm64`

**Comment by TheOtherBrian1** [comment]
I looked into issue regarding the inaccurate error messages. 

In September of 2022 last year, a [commit](https://github.com/supabase/pg_net/commit/6b8fcf24a46e28bdf99cfec43d027377f1ff5f01) was made to [worker.c](https://github.com/supabase/pg_net/blob/8b55e96d0d4b194dd8896b6f2a0fa8ef525aa8b3/src/worker.c#L467C9-L467C10) that incorporated the version number into the return message, i.e. the message went from "pg_net worker" to "pg_net VERSION worker". The [net.check_worker_is_up](https://github.com/supabase/pg_net/blob/8b55e96d0d4b194dd8896b6f2a0fa8ef525aa8b3/sql/pg_net.sql#L23) function was not updated to reflect this change. 

This issue can be resolved in 3 ways:

1. Remove the version number from the worker.c return message
2. Hard code the return number into the net.check_worker_is_up function
3. Change the comparison in the net.check_worker_is_up function to be general purpose for all new versions.

I prefer the third option because it would prevent this type of issue in the future. I took the liberty of rewriting the [offending line](https://github.com/supabase/pg_net/blob/8b55e96d0d4b194dd8896b6f2a0fa8ef525aa8b3/sql/pg_net.sql#L23C75-L23C75):

**Old Comparison:**

`... where backend_type = 'pg_net worker') ...`

**New Comparison:**

`... where backend_type ~ 'pg_net \d+(\.\d+){0,2} worker')`

As for why your requests are idling, I'm not particularly sure. That's a more complex issue that is not really something I can resolve, at least not quickly. Realistically, the extension should provide information about incomplete requests, but that feature isn't as robust. I'm not sure how effective this would be, but it might be helpful to monitor network traffic in your container using a terminal tool, such as _nload_, to see if the extension is making any requests.

I suspect your specific issue may have to do with your Docker configuration and its permission to access the internet.

**Comment by bentron2000** [comment]
Thanks @TheOtherBrian1 

I should have thought the timeout would still function in the case that network conditions/config were the issue here.
At the least, some sort of error messaging 🤔

I have configured the docker container to have internet access (as far as I can tell - i.e. I can curl things from a bash command line within the docker container)

Unsure where to turn now :/

Thanks again

**Comment by TheOtherBrian1** [comment]
@bentron2000 

I spent some time looking through the extensions code, and it seems that many of the errors are written to the PostgreSQL log file. On your set up, it may be located in the _var/log/postgresql_ directory. You should try viewing it if you haven't. If does show the cause, I would be very curious to know what it is.

**Comment by bentron2000** [comment]
@TheOtherBrian1 

Thanks for looking into this. You've really pointed me in a few directions that helped me today.

It appears as though the devil is in the detail here:
In my repro instructions I put '_create a database_' - this is the key issue. 

`pg_net` is does not work (or is not configured to work) on any database other than the `postgres` database that is present by default in this docker image.

When I install the extension and add a request in the existing postgres database - I have no issues with `pg_net` - (other than the previously noted bug in `check_worker_is_up()`)

Curiously though - there are no logs where you indicated.

Maintainers: The documentation should either:
1. Note that `pg_net` only works with the specific config present in the existing 'postgres' database (and why), or, 
2. Give instructions on how to configure it for an arbitrary database.

**Comment by TheOtherBrian1** [comment]
@bentron2000 

Wow, you are right, it is by default setup only to work on the _postgres_ database, as coded in **[Line 41 in worker.c](https://github.com/supabase/pg_net/blob/8b55e96d0d4b194dd8896b6f2a0fa8ef525aa8b3/src/worker.c#L41)**

`char* database_name = "postgres";`

Unfortunately, there were no hints in the documentation alluding to extension configuration. The documentation most certainly should be updated.

In the [_PG_Init function in worker.c](https://github.com/supabase/pg_net/blob/8b55e96d0d4b194dd8896b6f2a0fa8ef525aa8b3/src/worker.c#L448), there is the following code:

```
DefineCustomStringVariable(
      "pg_net.database_name", 
       "database where the pg_net worker is connected",
        NULL,
        &database_name,
        "postgres",
        PGC_SIGHUP, 0,
        NULL, NULL, NULL);
}

```


`DefineCustomStringVariable` comes from the [GUC.c library in PostgreSQL](https://github.com/postgres/postgres/blob/cd5f2a357014d5ed8205191f8e1fb180f1439599/src/backend/utils/misc/guc.c), which stands for Grand Unified Configuration. This code makes it possible to set the _database_name_ variable in the postgres.conf file.

Looking further into the code, it seems that the extension creates 3 configurable variables:

1. **pg_net.batch_size** (default: 200): An integer that limits the max number of rows that the extension will process from _net.http_request_queue_ during each read
2. **pg_net.ttl** (default: 6 hours): An interval that defines the max time a row in the _net._http_response_  will live before being deleted
3. **pg_net.database_name** (default: 'postgres'): A string that defines which database the extension is applied to

All these variables can be viewed with the following command:

`select 
    *
  from pg_settings WHERE name LIKE 'pg_net%'`

The _postgres.conf,_ file can be found with the following SQL command:

`SHOW config_file;`

I'm not too familiar with modifying the _postgres.conf_, so I Google searched for some hints on how to handle it properly.

I found a supabase post called  [Protecting reserved roles with PostgreSQL Hooks](https://supabase.com/blog/roles-postgres-hooks) that included a section about modifying configuration variables for a "supautils.c" extension:

_postgresql.conf_
```
# set the extension as preloaded, this will require a restart
shared_preload_libraries="supautils"

# the reserved roles
supautils.reserved_roles="supabase_storage_admin, supabase_auth_admin"
```

Extrapolating from that example, my guess is that you could modify your _postgresql.conf_ file with the following line(s) to get _pg_net_ to work:

```
#The below line may be necessary
#shared_preload_libraries="pg_net" 

pg_net.database_name=<"Your_new_database">
```

You may also be able to change the variable in SQL by using an `ALTER SYSTEM` query, as outlined by this [stackoverflow post](https://dba.stackexchange.com/questions/220700/how-to-edit-postgresql-conf-with-pgadmin-4)

After saving the file, you can execute `select pg_reload_conf()` to update _postgres.conf_ for your database. If the extension does not respond to the update, it may be necessary to restart your entire server.

If this solves your issue, I'll add some configuration guidelines to the readme.md for future readers.

**Comment by bentron2000** [comment]
@TheOtherBrian1 

Amazing! That certainly looks like the information required to get this working. 

I made a quick attempt to use `ALTER SYSTEM` to modify the config but the postgres user in the image does not have sufficient permissions for this purpose.

However, adding `pg_net.database_name=<"Your_new_database">` to the `postgresql.conf` file does work - and `pg_net` operates correctly once this is set.

This is the solution :) Really appreciate the time you've spent here assisting me.

**To summarize:**
When using `pg_net` with a database named other than `postgres` - the variable `pg_net.database_name` must be set to the name of the database the worker should attach to. This may be able to be done at runtime - however, is best achieved via `postgresql.conf`.
NB. This suggests that only one instance of `pg_net` can be enabled per postgres instance.

Thanks again :) B

**Comment by soedirgo** [comment]
Thanks all, and sorry for the confusion! I'm updating the README [here](https://github.com/supabase/pg_net/pull/99) to clarify on this.
